packages/@vue/cli/lib/Generator.js

```diff
@@ -242,12 +275,22 @@ module.exports = class Generator {
     debug('vue:cli-files')(this.files)
   }
 
-  hasPlugin (_id) {
+  hasPlugin (_id, _version) {
     return [
       ...this.plugins.map(p => p.id),
-      ...Object.keys(this.pkg.devDependencies || {}),
-      ...Object.keys(this.pkg.dependencies || {})
-    ].some(id => matchesPluginId(_id, id))
+      ...this.otherPlugins
+    ].some(id => {
+      if (!matchesPluginId(_id, id)) {
+        return false
+      }
+
+      if (!_version) {
+        return true
+      }
+
+      const version = this.pkg.dependencies[id] || this.pkg.devDependencies[id]
```

Comment:
This typically returns a semver range like `^3.9.0` and is not a valid version to be used as the first argument of `semver.satisfies`.
Better use [getInstalledVersion](https://github.com/vuejs/vue-cli/blob/4af27247ccabe2ba5da57b6f216d17ad0767b8a8/packages/%40vue/cli/lib/util/ProjectPackageManager.js#L151) and do some workaround in the unit tests.